src/memfs.ts

```diff
@@ -233,6 +232,11 @@ export class InMemoryFileSystem implements ts.ParseConfigHost, ts.ModuleResoluti
 	trace(message: string) {
 		this.logger.log(message);
 	}
+
+	// TODO: move this method to util and use through all the code
+	private path2uri(path: string): string {
```

Comment:
new method is used in `readFile()` and `fileExists()`. `util.resolve()` indeed looks at strict mode (expected POSIX-style resolution) and host OS (On Windows we expect to use Windows-specific resolution unless we are working in strict mode). `url.resolve()` probably may fail if OS-specific path is passed